libraries/standard/mqtt/include/types/iot_mqtt_types.h

```diff
@@ -717,197 +835,146 @@ typedef struct IotMqttConnectInfo
-             *
+             * @brief MQTT PUBLISH packet deserializer function.
+             * @param[in,out] _mqttPacket* Pointer to an MQTT packet structure
+			 *
              * <b>Default implementation:</b> #_IotMqtt_DeserializePublish
              */
-            IotMqttError_t ( * publish )( struct _mqttPacket * /* pPublish */ );
-
+            IotMqttDeserializer_t publish;
             /**
-             * @brief PUBACK packet deserializer function.
-             * @param[in,out] _mqttPacket* Pointer to an MQTT packet struct representing a PUBACK.
-             *
+             * @brief MQTT PUBACK packet deserializer function.
+             * @param[in,out] _mqttPacket* Pointer to an MQTT packet structure
+			 *
              * <b>Default implementation:</b> #_IotMqtt_DeserializePuback
              */
-            IotMqttError_t ( * puback )( struct _mqttPacket * pPuback );
-
+            IotMqttDeserializer_t puback;
             /**
-             * @brief SUBACK packet deserializer function.
-             * @param[in,out] _mqttPacket* Pointer to an MQTT packet struct representing a SUBACK.
-             *
+             * @brief MQTT SUBACK packet deserializer function.
+             * @param[in,out] _mqttPacket* Pointer to an MQTT packet structure
+			 *
              * <b>Default implementation:</b> #_IotMqtt_DeserializeSuback
              */
-            IotMqttError_t ( * suback )( struct _mqttPacket * /* pSuback */ );
-
+            IotMqttDeserializer_t suback;
             /**
-             * @brief UNSUBACK packet deserializer function.
-             * @param[in,out] _mqttPacket* Pointer to an MQTT packet struct representing an UNSUBACK.
-             *
+             * @brief MQTT UNSUBACK packet deserializer function.
+             * @param[in,out] _mqttPacket* Pointer to an MQTT packet structure
+			 *
              * <b>Default implementation:</b> #_IotMqtt_DeserializeUnsuback
              */
-            IotMqttError_t ( * unsuback )( struct _mqttPacket * /* pUnsuback */ );
-
+            IotMqttDeserializer_t unsuback;
             /**
-             * @brief PINGRESP packet deserializer function.
-             * @param[in,out] _mqttPacket* Pointer to an MQTT packet struct representing a PINGRESP.
-             *
+             * @brief MQTT PINGRESP packet deserializer function.
+             * @param[in,out] _mqttPacket* Pointer to an MQTT packet structure
+			 *
              * <b>Default implementation:</b> #_IotMqtt_DeserializePingresp
              */
-            IotMqttError_t ( * pingresp )( struct _mqttPacket * /* pPingresp */ );
+            IotMqttDeserializer_t pingresp;
         } deserialize; /**< @brief Overrides the packet deserialization functions for a single connection. */
     } IotMqttSerializer_t;
+
+/**
+ * @brief Utility macro for creating serializer override selector functions
+ */
+#define IOT_MQTT_SERIALIZER_OVERRIDE_SELECTOR( _funcType_t, _funcName, _defaultFunc, _serializerMember ) \
```

Comment:
This seems to be for internal use only.  Please move to `iot_mqtt_internal.h` (and rename per internal macro name).

Reply:
Good points, moved to `iot_mqtt_internal.h` and prefixed the name with `_`.